embedfs/embed_test.go

```diff
@@ -338,10 +325,188 @@ func TestJoin(t *testing.T) {
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			fs := New(&empty)
+			fs := New(testdata.GetTestData())
 
 			got := fs.Join(tc.path...)
 			assert.Equal(t, tc.want, got)
 		})
 	}
 }
+
+// Additional comprehensive tests using rich test data
+
+func TestEmbedfs_ComprehensiveOpen(t *testing.T) {
+	t.Parallel()
+	
+	fs := New(testdata.GetTestData())
+	
+	// Test opening existing embedded file with content
+	f, err := fs.Open("/testdata/file1.txt")
+	require.NoError(t, err)
+	assert.Equal(t, "testdata/file1.txt", f.Name())
+	require.NoError(t, f.Close())
+}
+
+func TestEmbedfs_ComprehensiveRead(t *testing.T) {
+	t.Parallel()
+	
+	fs := New(testdata.GetTestData())
+	
+	f, err := fs.Open("/testdata/file1.txt")
+	require.NoError(t, err)
+	defer f.Close()
+	
+	// Read the actual content
+	buf := make([]byte, 100)
+	n, err := f.Read(buf)
+	require.NoError(t, err)
+	assert.Equal(t, "Hello from embedfs!", string(buf[:n]))
+}
```

Comment:
This is redundant - `TestOpen` already opens and reads the contents of the files.

```suggestion
```